libs/sdk-core/src/invoice.rs

```diff
@@ -167,28 +172,35 @@ pub fn add_lsp_routing_hints(
     // We make sure the hint we add does not conflict with other hints.
     // The lsp hint takes priority so in case the lsp hop is already in one of the existing hints
     // We make sure not to include them in the new hints.
-    let unique_hop_hints: Vec<lightning::routing::router::RouteHint> = match lsp_hint {
-        None => invoice.route_hints(),
-        Some(lsp_hint) => match include_route_hints {
+    let unique_hop_hints: Vec<lightning::routing::router::RouteHint> = match route_hints.len() {
+        0 => invoice.route_hints(),
+        _ => match merge_with_existing {
             true => {
-                let mut all_hints: Vec<lightning::routing::router::RouteHint> = invoice
-                    .route_hints()
-                    .into_iter()
+                let unique_to_add: Vec<&RouteHint> = route_hints
```

Comment:
Maybe a better way to find `unique_to_add` is

```rust
let invoice_hints_hop_src_node_ids: Vec<String> = invoice
    .route_hints()
    .into_iter()
    .flat_map(|hint| hint.0)
    .map(|hop| hop.src_node_id.serialize().encode_hex::<String>())
    .collect();
let unique_to_add: Vec<&RouteHint> = route_hints
    .iter()
    .filter(|hint| {
        hint.hops
            .iter()
            .all(|hop| !invoice_hints_hop_src_node_ids.contains(&hop.src_node_id))
    })
    .collect();
```

Avoids the 4-level-deep nesting in the filter, which is error-prone.

Reply:
Done, thank you.